src/core/Target.mli

```diff
@@ -63,12 +63,20 @@ type lockfile = {
 (** A lockfile to be used during matching. See also {!Lockfile_xtarget.t}, an
     augmented version with the contents of the lockfile. *)
 
+(** A scan mode that tells us whether we need to do both Intra-file
+    and Inter-file scans or just an Inter-file scan. When performing
```

Comment:
When do we need to do *both* an intra-file and an inter-file scan? Maybe it makes sense when I look into the Pro PR, but this data type even with the comment is not very clear to me.